### ISSUE TITLE
Add Support for using AAD OAuth2.0 Authorization

### ISSUE THREAD

**Issue by PlagueHO** [PlagueHO/CosmosDB#427]
Cosmos DB now supports AAD OAuth 2.0 Authorization: https://docs.microsoft.com/en-us/rest/api/cosmos-db/access-control-on-cosmosdb-resources

Enable support for this.

**Comment by ArmaanMcleod** [comment]
@PlagueHO I'm happy to help with this one. I've been wanting to use AAD tokens instead of master keys for some time 😄.

Was thinking of having a way to pass in the AAD access token from `Get-AzAccessToken`:

```powershell
$aadToken = (Get-AzAccessToken -ResourceUrl 'https://<cosmosDBAccountName>.documents.azure.com').Token | ConvertTo-SecureString -AsPlainText -Force
```

But not sure how could be passed into `New-CosmosDBContext`. I looked at `-Token` but this seems to be mainly used for resource tokens. Would AAD tokens need a new parameter?

**Comment by PlagueHO** [comment]
@ArmaanMcleod - this is now available in the [5.0.0-preview0001](https://www.powershellgallery.com/packages/CosmosDB/5.0.0-preview0001). I'll close this for now, but open new issue if need some changes. 

I implemented `New-CosmosDbContextAADToken` but called `Get-CosmosDbContextEntraIdToken` - see the README.md
